### DISCUSSION TITLE
Test Discussion!

### DISCUSSION THREAD

**Discussion by raytiley** [trms/wp-cablecast#33] [General]
Hey @bryanharley  - if someone had a question about using this plugin, as long as I get #29 working you would help them out?

**Comment by bryanharley** [comment]
Sure thing! I’d help them out regardless.

**Comment by petervangorder** [comment]
Is this plugin up-to-date and supported?

I'm seeing some 404's on linked code like this one:
https://developer.wordpress.org/plugins/cron/hooking-into-the-system-task-scheduler/

I've installed it and added this code:
https://gist.github.com/bryanharley/7986451176db08f982683c1b973abc8c

But see no shows, nada.

Thanks in advance!

**Reply by bryanharley** [reply]
Hi Peter.  I will fix the link, here's the new one: https://developer.wordpress.org/plugins/cron/hooking-wp-cron-into-the-system-task-scheduler/

Once you've installed the plugin, entered your Cablecast address, and set up a scheduler for the WordPress cron - you should start to see Cablecast shows sync over into WordPress. Depending on how many shows you have, this initial syncing process could take a long time. In your WordPress admin dashboard, click on "Shows" in the left navigation. This is where you can see how many shows have synced so far.  

The code you linked to is optional, not required for the plugin to work.  Where ever you put that code it will display which show is currently scheduled and which one is coming up next for a given channel.  For example, I placed this on my page that displays the schedule for one of our channels: [cmac.tv/public](https://cmac.tv/public).

Let me know if you have any more questions.

**Comment by petervangorder** [comment]
Wow, thanks for the quick reply!
It was about an hour to 45 min between the time I installed the plugin and when I contact you - nothing appeared in the "Shows" list.

I then followed the above instructions, and set the cron job to 5 minuets (10 minutes ago) and still no shows.   We do have a lot of shows, but I thought I'd see something by now, but I haven't.   Dang.

**Reply by bryanharley** [reply]
Couple things to confirm.  

1. Have you entered the URL of your Cablecast server on the Cablecast Settings page in the WP admin dashboard?  This should be the URL that takes you to FrontDoor and is accessible on the public internet.

2. Have you disabled the WP-Cron by adding this line to your wp-config.php file?  `define( 'DISABLE_WP_CRON', true );`

3. Which method did you use for scheduling the WP-Cron manually?  Can you confirm it is running as scheduled?  If you're having difficulty, I've used this free website to setup the cron job: https://cron-job.org.  

Once these 3 things are setup correctly, each time the cron runs, you should see shows synced into WordPress on the Shows page.

**Comment by petervangorder** [comment]
1) Yes.  (I even tried a "sister station's" url)
2) Yes.
3) I tried scheduling the cron job via the cPanel and with my new account on https://console.cron-job.org/

Then I go to this page:
https://dream-create-communicate.com/OVTV/wp-admin/edit.php?post_type=show
and look for show records to show up - but none ever show up.

**Reply by raytiley** [reply]
Hi Peter,

Can you give us a screenshot of the cablecast settings page above your server url? It should look something like this: 

![image](https://github.com/trms/wp-cablecast/assets/41680/1859df81-370e-432d-aa7d-0ce0f5329b94)

**Reply by petervangorder** [reply]
Sure sure!

<img width="629" alt="Screenshot 2024-02-26 at 3 48 09 PM" src="https://github.com/trms/wp-cablecast/assets/3176354/e8a7060c-4552-482c-8308-1d2bff56acda">

**Reply by raytiley** [reply]
@petervangorder - That server url is not a Cablecast server. I would support@cablecast.tv and we might be able to help get you sorted.

**Comment by petervangorder** [comment]
You help just solve it!
I was being told to use "https://okemovalley.tv"

But as soon as you said it wasn't, I dug deeper and found:  http://lpctvcontrol.com
I'm up and running now!  Thank you!

Where do I go to buy you a "cup of coffee" - so to speak?

**Comment by petervangorder** [comment]
Good afternoon Ray,

2 Things....

1) I was serious about pushing your some money for your help (and I honestly hope I'm not violating a GitHub policy by offering), and whereas I said "coffee", I wasn't thinking of sending $3.50, but more like $35.  Though perhaps that's just as insulting - I apologize if it is.  If not, let me know how to PayPal you some money, as my functional gratitude.

2) The cron ran once and it successfully loaded 100 shows - but then no more.  
I'm using cron-job.org and I'm seeing SUCCESS messages from its runs.  So I'm confused.

**Comment by raytiley** [comment]
> 1. I was serious about pushing your some money for your help (and I honestly hope I'm not violating a GitHub policy by offering), and whereas I said "coffee", I wasn't thinking of sending $3.50, but more like $35.  Though perhaps that's just as insulting - I apologize if it is.  If not, let me know how to PayPal you some money, as my functional gratitude.

No need. I get paid by Cablecast already so it's kinda part of my job to help out here :)





> 2\. The cron ran once and it successfully loaded 100 shows - but then no more.
>     I'm using cron-job.org and I'm seeing SUCCESS messages from its runs.  So I'm confused.

What does the status say in the cablecast settings?

**Comment by petervangorder** [comment]
Paid by Cablecast - Ah!   Well, thanks for all your support just the same!

**Status:**
Successful
200 OK

**Details:**
Server: Sucuri/Cloudproxy
Date: Tue, 27 Feb 2024 19:45:00 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 0
Connection: keep-alive
X-Sucuri-ID: 15030
X-XSS-Protection: 1; mode=block
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
Content-Security-Policy: upgrade-insecure-requests;
Expires: Wed, 11 Jan 1984 05:00:00 GMT
Cache-Control: no-cache, must-revalidate, max-age=0
Vary: Accept-Encoding
X-Sucuri-Cache: HIT

**Reply by bryanharley** [reply]
In the WordPress admin dashboard, click "Cablecast Settings" on the left side.  At the top of the page it should say: "There are X remaining shows out of X shows updated after X".  Can you paste what that says?

**Reply by petervangorder** [reply]
"There are 0 remaining shows out of 0 shows updated after"

**Comment by petervangorder** [comment]
Patrick Cody just filled me in - that you did some custom coding for our Drupal platform.
We need to get off Drupal - that's why I'm doing this investigation into Wordpress.

Long story short - Patric just said that if we need/could hire you to help us transition (not build the Wordpress site, I'd be doing that) we have the budget for that.    Just FYI.

**Comment by petervangorder** [comment]
Hi,
Just wondering...   Has this plugin been abandoned?

**Reply by raytiley** [reply]
@petervangorder  - Abandoned is a strong word :)

A bunch of people are using it and having success with it, but there isn't. a lot of effort from Cablecast staff members. I have some ideas to make it easier to use and take advantage of some newer features, but probably won't get any major attention till later this year.

**Comment by petervangorder** [comment]
Thanks Raytiley

Then I guess I just wish there were some/any error codes that would tell my why it only imported the first 100 - and then stopped and won't import anything ever again.
My cron job reports a success, I don't see any other errors.

**Comment by raytiley** [comment]
@petervangorder - Are you able to open a support request with support@cablecast.tv We don't have support staff monitoring this repo's issues, its mostly for developers / pr's etc. When troubleshooting we might need sensitive info that we don't want to put into public discussion.

Thanks
